python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -203,7 +203,12 @@ def download_thumbnail(url, bundle):
                 os.chmod(path_to_cached_thumb, 0666)
             finally:
                 os.umask(old_umask)
-
+        else:
+            # Update access and modified time to "now" so the file will be kept
+            # around when culling old files in the cache.
+            # `_get_thumbnail_path` returns a full path with the extension if
+            # the thumb exists.
+            os.utime(path_to_cached_thumb, None)
```

Comment:
maybe, for clarity, we could have this in a method that is being called? something like - `self._indicate_resource_accessed()` or something?